### ISSUE TITLE
Websocket ping - (lib/Web.js)

### ISSUE THREAD

**Issue by rfpaula** [howdyai/botkit#1375]
Hi, It's all right?

I think that there are a compatibility problem between **_botkit 0.6.14_** and **_ws 4.0.0_** and newer versions.

Until the **version 3.3.3 of ws** the signature of ping method was **ping (data, mask, failSilently)**, on the **version 4.0.0** the signature was modified to **ping (data, mask, cb)**.

How the botkit version 0.6.14 on the **_lib/Web.js_** set a setInterval function that use _ws.ping('', false, true)_ , with the package.json dependency setup as **"ws": "^5.1.1"**, happens a error when this code run because the new signature is **ping (data, mask, cb)** incompatible with the signature used in setinterval function.

Bellow the related code.

## **botkit/lib/Web.js 0.6.14**
`
botkit/lib/Web.js

        var interval = setInterval(function ping() {
            wss.clients.forEach(function each(ws) {
                if (ws.isAlive === false) {
                    return ws.terminate();
                }
                //  if (ws.isAlive === false) return ws.terminate()
                ws.isAlive = false;
                ws.ping('', false, true);
            });
        }, 30000);

`

## ** ws 3.3.3**
`
ws/lib/WebSocket.js

  /**
   * Send a ping message.
   *
   * @param {*} data The message to send
   * @param {Boolean} mask Indicates whether or not to mask `data`
   * @param {Boolean} failSilently Indicates whether or not to throw if `readyState` isn't `OPEN`
   * @public
   */
  ping (data, mask, failSilently) {
    if (this.readyState !== WebSocket.OPEN) {
      if (failSilently) return;
      throw new Error('not opened');
    }

    if (typeof data === 'number') data = data.toString();
    if (mask === undefined) mask = !this._isServer;
    this._sender.ping(data || constants.EMPTY_BUFFER, mask);
  }

`

## **a206e98  on 5 Jan ws 4.0.0**
`
ws/lib/WebSocket.js

  /**
   * Send a ping.
   *
   * @param {*} data The data to send
   * @param {Boolean} mask Indicates whether or not to mask `data`
   * @param {Function} cb Callback which is executed when the ping is sent
   * @public
   */
  ping (data, mask, cb) {
    if (typeof data === 'function') {
      cb = data;
      data = mask = undefined;
    } else if (typeof mask === 'function') {
      cb = mask;
      mask = undefined;
    }

    if (this.readyState !== WebSocket.OPEN) {
      const err = new Error(
        `WebSocket is not open: readyState ${this.readyState} ` +
          `(${readyStates[this.readyState]})`
      );

      if (cb) return cb(err);
      throw err;
    }

    if (typeof data === 'number') data = data.toString();
    if (mask === undefined) mask = !this._isServer;
    this._sender.ping(data || constants.EMPTY_BUFFER, mask, cb);
  }

`

Best regards,
Rodrigo Fernandes de Paula

**Comment by peterswimm** [comment]
Are you still seeing these troubles?

**Comment by rfpaula** [comment]
Hi Peter,
Because of this issue i fixed the botkit version at 0.6.13, the last one that has ws ^3.x.x as dependency, then i don't have problems any more, but i can't update my botkit package.

I checked the ping ws.ping method call on Web.js at the last version (0.6.16) and the method signature is the same ws.ping('', false, true) as so the ws dependency (^5.x.x). I did not do a new test, but i belive that the problem persists.

Best regards,
Rodrigo Fernandes de Paula
